internal/formatter/yaml_formatter.go

```diff
@@ -31,5 +31,11 @@ func YAMLFormatter(content []byte) ([]byte, bool, error) {
 		return nil, false, errors.Wrap(err, "marshalling YAML node failed")
 	}
 	formatted := b.Bytes()
+
+	// required to preserve yaml files starting with "---" as yaml.Encoding strips them
+	if bytes.HasPrefix(content, []byte("---\n")) {
```

Comment:
should it be checked that `formatted` does not contain prefix "---\n" too? 

```suggestion
	if !bytes.HasPrefix(formatted, []byte("---\n") && bytes.HasPrefix(content, []byte("---\n")) {
```

Reply:
From my tests it is always stripped (there the need for this workaround), but makes no harm to add it just in case 👍